tests/test_chat.py

```diff
@@ -640,3 +640,133 @@ def test_form_prompt_string_assistant_content_before_tool_calls_responses() -> N
         "Assistant:"
     )
     assert form_prompt_string(messages) == expected
+
+
+def test_form_prompt_string_with_instructions_responses() -> None:
+    """Test formatting with developer instructions in responses format."""
+    messages = [
+        {"role": "user", "content": "What can you do?"},
+    ]
+    expected = "System: Always be concise and direct in your responses.\n\n" "User: What can you do?\n\n" "Assistant:"
+    assert form_prompt_string(messages, instructions="Always be concise and direct in your responses.") == expected
+
+
+def test_form_prompt_string_with_instructions_and_tools_responses() -> None:
+    """Test formatting with developer instructions and tools in responses format."""
+    messages = [
+        {"role": "user", "content": "What can you do?"},
+    ]
+    tools = [
+        {
+            "type": "function",
+            "name": "search",
+            "description": "Search the web for information",
+            "parameters": {
+                "type": "object",
+                "properties": {"query": {"type": "string", "description": "The search query"}},
+                "required": ["query"],
+            },
+            "strict": True,
+        }
+    ]
+    expected = (
+        "System: Always be concise and direct in your responses.\n\n"
+        "You are a function calling AI model. You are provided with function signatures within <tools> </tools> XML tags. "
```

Comment:
how come in this unit test, the system/developer message is appearing before the tool-definitions, but in the other tests, their order is reversed?  

I think we previously discussed/investigated the optimal order (dont recall), let's stick with that order consistently.

To me this intuitively seems like the better order:

```
System:  You are ACME Support, ...
You are a function calling AI model. ...
```

but in other unit tests, it appears like you are putting the tool definitions first.